data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java

```diff
@@ -13,50 +13,77 @@
-import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.spec.datastructures.metadata.StateAndMetaData;
+import tech.pegasys.teku.spec.datastructures.state.Checkpoint;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
 
-public class GetStateFinalityCheckpoints extends AbstractHandler implements Handler {
+public class GetStateFinalityCheckpoints extends AbstractGetSimpleDataFromState {
   private static final String OAPI_ROUTE = "/eth/v1/beacon/states/:state_id/finality_checkpoints";
   public static final String ROUTE = routeWithBracedParameters(OAPI_ROUTE);
-  private final ChainDataProvider chainDataProvider;
 
-  public GetStateFinalityCheckpoints(
-      final DataProvider dataProvider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
-    this.chainDataProvider = dataProvider.getChainDataProvider();
+  private static final SerializableTypeDefinition<BeaconState> DATA_TYPE =
+      SerializableTypeDefinition.object(BeaconState.class)
+          .withField(
+              "previous_justified",
+              Checkpoint.SSZ_SCHEMA.getJsonTypeDefinition(),
+              BeaconState::getPreviousJustifiedCheckpoint)
+          .withField(
+              "current_justified",
+              Checkpoint.SSZ_SCHEMA.getJsonTypeDefinition(),
+              BeaconState::getCurrentJustifiedCheckpoint)
+          .withField(
+              "finalized",
+              Checkpoint.SSZ_SCHEMA.getJsonTypeDefinition(),
+              BeaconState::getFinalizedCheckpoint)
+          .build();
+
+  private static final SerializableTypeDefinition<StateAndMetaData> RESPONSE_TYPE =
+      SerializableTypeDefinition.object(StateAndMetaData.class)
+          .withField("data", DATA_TYPE, StateAndMetaData::getData)
+          .build();
+
+  public GetStateFinalityCheckpoints(final DataProvider dataProvider) {
+    this(dataProvider.getChainDataProvider());
   }
 
-  GetStateFinalityCheckpoints(
-      final ChainDataProvider chainDataProvider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
-    this.chainDataProvider = chainDataProvider;
+  GetStateFinalityCheckpoints(final ChainDataProvider chainDataProvider) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getStateFinalityCheckpoints")
+            .summary("Get state finality checkpoints")
+            .description(
+                "Returns finality checkpoints for state with given 'state_id'. In case finality is not yet achieved, checkpoint should return epoch 0 and ZERO_HASH as root.")
+            .tags(TAG_BEACON)
+            .pathParam(PARAMETER_STATE_ID)
+            .response(SC_OK, "Request successful", RESPONSE_TYPE)
+            .response(SC_NOT_FOUND, "Not found", HTTP_ERROR_RESPONSE_TYPE)
```

Comment:
should be able to clean up this with the patch I gave you, then the endpoints will be consistent for not_found